### PR TITLE
BB AI QA testing fixes

### PR DESCRIPTION
**Pull request by shogunpurple**
## Description
Just fixes for some bugs and issues I found while doing QA on the AI functionality for 3.0

- Default cloud config didn't show up when you didn't have custom configs configured, accounted for
- OpenAI automation step doesn't appear on cloud as we restricted it to self host. Now shows when you have Budibase AI enabled
- Quotas weren't updating properly for budibase AI credits and custom configs, fixed (along with work in https://github.com/Budibase/budibase-pro/pull/327)
- Fix the "Clean data" AI function

### CODE REVIEW

File: packages/builder/src/pages/builder/portal/settings/ai/index.svelte [hunk 1]
```diff
@@ -32,9 +32,7 @@
   async function fetchAIConfig() {
     try {
       const aiDoc = await API.getConfig(ConfigTypes.AI)
```
Comment (adrinr) [inline packages/builder/src/pages/builder/portal/settings/ai/index.svelte:34]:
```suggestion
      fullAIConfig = await API.getConfig(ConfigTypes.AI)
```

File: packages/worker/src/api/controllers/global/configs.ts [hunk 2]
```diff
@@ -44,9 +44,7 @@ const getEventFns = async (config: Config, existing?: Config) => {
       fns.push(events.email.SMTPCreated)
     } else if (isAIConfig(config)) {
       fns.push(() => events.ai.AIConfigCreated)
-      fns.push(() =>
-        pro.quotas.updateCustomAIConfigCount(Object.keys(config.config).length)
-      )
+      fns.push(() => pro.quotas.addCustomAIConfig)
```
Comment (adrinr) [inline packages/worker/src/api/controllers/global/configs.ts:47]: Is this meant to be `      fns.push(() => pro.quotas.addCustomAIConfig())`

File: packages/server/src/automations/actions.ts [hunk 3]
```diff
@@ -89,21 +90,27 @@ export const BUILTIN_ACTION_DEFINITIONS: Record<
...
   if (await features.flags.isEnabled(FeatureFlag.AUTOMATION_BRANCHING)) {
     BUILTIN_ACTION_DEFINITIONS["BRANCH"] = branch.definition
   }
+  if (
+    env.SELF_HOSTED ||
+    (await features.flags.isEnabled(FeatureFlag.BUDIBASE_AI)) ||
+    (await features.flags.isEnabled(FeatureFlag.AI_CUSTOM_CONFIGS))
+  ) {
+    BUILTIN_ACTION_DEFINITIONS["OPENAI"] = openai.definition
+    // @ts-ignore
```
Comment (melohagan) [inline packages/server/src/automations/actions.ts:110]: NAB: Why is this needed>

File: packages/worker/src/api/controllers/global/configs.ts [hunk 4]
```diff
@@ -342,29 +337,43 @@ export async function find(ctx: UserCtx) {
...
+  if (type === ConfigType.OIDC_LOGOS) {
+    enrichOIDCLogos(config)
+  } else if (type === ConfigType.AI) {
+    await handleAIConfig(config)
+  }
+}
+
+async function handleAIConfig(config: AIConfig) {
+  await pro.sdk.ai.enrichAIConfig(config)
+  stripApiKeys(config)
```
Comment (melohagan) [inline packages/worker/src/api/controllers/global/configs.ts:366]: NAB: Ideally you wouldn't need to worry about this? Is there a risk of ChatGPT keys being exposed if this line wasn't here?

**Review by melohagan — APPROVED** [review]
LGTM

File: packages/worker/src/api/controllers/global/configs.ts (hunk 4 above)
Comment (shogunpurple) [reply packages/worker/src/api/controllers/global/configs.ts:366]: Yep, it's so they don't get returned to the frontend when you pull the configs

**Review by mike12345567 — APPROVED** [review]
